client/src/main/java/io/pravega/client/netty/impl/RawClient.java

```diff
@@ -131,12 +140,27 @@ private void closeConnection(Throwable exceptionToInflightRequests) {
     }
 
     public <T extends Request & WireCommand> CompletableFuture<Reply> sendRequest(long requestId, T request) {
+        return sendRequest(requestId, request, this.connectionTimeout);
+    }
+
+    public <T extends Request & WireCommand> CompletableFuture<Reply> sendRequest(long requestId, T request,
+                                                                                  Duration timeout) {
         return connection.thenCompose(c -> {
             log.debug("Sending request: {}", request);
             CompletableFuture<Reply> reply = new CompletableFuture<>();
             synchronized (lock) {
                 requests.put(requestId, reply);
             }
+
+            Futures.onTimeout(reply, timeout, connectionFactory.getInternalExecutor(), ex -> {
+                synchronized (lock) {
+                    log.debug("Request {} times out after {}", requestId, timeout.toString());
```

Comment:
Can we log it @info/@warn level since we can indicate that server took greater 30 seconds?